### DISCUSSION TITLE
Meaning and values of handlers 0x04AF and 0x04AE

### DISCUSSION THREAD

**Discussion by KLiebrecht1** [emsesp/EMS-ESP32#1462] [General]
ems-esp Version 3.6.3

Device HM200
Device-ID 0x53

Handler 0x04AF - Heatpump Energy Consumption - 4 Byte - 1/100 kWh
006+02:35:19.912 N 17: [emsesp] heatpump(0x53) -B-> All(0x00), ?(0x04AF), data: 00 02 85 22 00 00 00 00 00 00 00 00 (offset 24)
00 02 85 22 = 165154 = 1651.54 kWh

Handler 0x04AE - Heatpump Energy Produced - 4 Byte - 1/100 kWh
006+02:30:18.527 N 7: [emsesp] heatpump(0x53) -B-> All(0x00), ?(0x04AE), data: 00 08 8C 93 00 00 00 00 00 00 00 00 (offset 4)
00 08 8C 93 = 560275 = 5602.75 kWh

Can you please integrate them?

**Comment by proddy** [comment]
Could you create custom entities to test first, then post a screen shot. It'll help us quickly add them (https://emsesp.github.io/docs/Troubleshooting/#im-missing-certain-data-from-an-ems-device)

**Reply by KLiebrecht1** [reply]
Sure...
![Custom_entities](https://github.com/emsesp/EMS-ESP32/assets/89344375/16ca2a99-e82d-4644-acd2-e6459a792aa7)

**Reply by proddy** [reply]
perfect!

**Reply by MichaelDvP** [reply]
Hmm we had this in #1349 and #1350 for main heatpump (device-id 8) already implemented, but with different meaning. 
What is right? I don't think the telegram is different when coming from 0x08 or 0x53.

**Reply by KLiebrecht1** [reply]
Maybe its because our system is brand new. Installed 2 months ago.

Buderus Boiler Logano plus GB212-30
Buderus Hybrid Manager HM200.2
Buderus Heatpump Logatherm WLW196i.2 AR

[emsesp_info.json](https://github.com/emsesp/EMS-ESP32/files/13466253/emsesp_info.json)

**Reply by MichaelDvP** [reply]
Please try https://github.com/MichaelDvP/EMS-ESP32/releases/tag/test
I've added the entities from boiler to heatpump. For now only what we already have, check how offset 24 fit's in this scheme, than i'll add it.
You have a hybrid system with heatpump at id 53, the single heatpumps are on id 08, you can't read the these telegrams from your boiler at 08. I need to add them in the heatpump class.

**Reply by KLiebrecht1** [reply]
Looks good!

![Bildschirmfoto vom 2023-11-26 12-53-03](https://github.com/emsesp/EMS-ESP32/assets/89344375/eae87815-4039-4d6d-91f0-afe73b7fc266)
